src/core/PathUtil.cpp

```diff
@@ -0,0 +1,139 @@
+#include "PathUtil.h"
+
+#include <QDebug>
+#include <QDir>
+#include <QFileInfo>
+
+#include "ConfigManager.h"
+
+namespace PathUtil
+{
+	Base relativeBases[] = { ProjectDirBase, FactorySampleBase, UserSampleBase, UserVSTBase, PresetBase,
+		UserLADSPABase, DefaultLADSPABase, UserSoundfontBase, DefaultSoundfontBase, UserGIGBase, DefaultGIGBase };
+
+	QString baseLocation(const Base & base)
+	{
+		QString loc = "";
+		switch (base)
+		{
+			case ProjectDirBase       : loc = ConfigManager::inst()->userProjectsDir(); break;
+			case FactorySampleBase    :
+			{
+				QDir fsd = QDir(ConfigManager::inst()->factorySamplesDir());
+				loc = fsd.absolutePath() + "/"; break;
```

Comment:
Is there a reason for appending `"/"` here? `QDir::cleanPath` will remove it later.

Reply:
I think I wanted it there to signify that it's a path, not a file? If it's unnecessary I can remove it.

Reply:
```suggestion
				loc = fsd.absolutePath(); break;
```